chain-storage-sqlite-old/src/lib.rs

```diff
@@ -460,18 +450,72 @@ where
             .transaction()
             .map_err(|err| Error::BackendError(Box::new(err)))?;
 
-        Self::get_nth_ancestor_internal(&tx, block_hash, distance)
+        get_nth_ancestor_internal::<B>(&tx, block_hash, distance)
     }
+}
 
-    fn get_nth_ancestor_internal(
-        connection: &Connection,
-        block_hash: &B::Id,
-        distance: u64,
-    ) -> Result<BlockInfo<B::Id>, Error> {
-        for_path_to_nth_ancestor_internal::<B, _>(connection, block_hash, distance, |_| {})
+fn get_block_info_internal<B>(
+    connection: &Connection,
+    block_hash: &B::Id,
+) -> Result<BlockInfo<B::Id>, Error>
+where
+    B: Block,
```

Comment:
Try making these generic over the hash type, then you probably won't need turbofishing at the use sites.